test/server/cards/02_SHD/leaders/HunterOutcastSergeant.spec.ts

```diff
@@ -0,0 +1,75 @@
+describe('Hunter, Outcast Sergeant', function () {
+    integration(function (contextRef) {
+        describe('Hunter\'s leader undeployed ability', function () {
+            it('should reveal a resource and bring back to hand if it share a name with a friendly unit in play and resource the top card', function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['battlefield-marine', 'echo#restored'],
+                        spaceArena: ['millennium-falcon#piece-of-junk'],
+                        resources: ['millennium-falcon#landos-pride', 'echo#restored', 'battlefield-marine', 'devotion'],
+                        leader: 'hunter#outcast-sergeant',
+                    },
+                    player2: {
+                        resources: ['echo#restored']
+                    }
+                });
+
+
+                const { context } = contextRef;
+                const set2Falcon = context.player1.findCardByName('millennium-falcon#landos-pride', 'resource');
+                const resourceEcho = context.player1.findCardByName('echo#restored', 'resource');
+
+                context.player1.clickCard(context.hunter);
+
+                // only cards which share a name with friendly unique unit
+                expect(context.player1).toBeAbleToSelectExactly([set2Falcon, resourceEcho]);
+                expect(context.player1).toHaveChooseNoTargetButton();
+
+                context.player1.clickCard(set2Falcon);
+
+                expect(context.player2).toBeActivePlayer();
+                expect(context.player1.resources.length).toBe(4);
```

Comment:
Can we declare some deck cards explicitly so we can confirm that the specific card is moved? Same for deployed test